SplunkRumWorkspace/SplunkRum/SplunkRum/NotificationEvents.swift

```diff
@@ -19,18 +19,23 @@ import Foundation
 import UserNotifications
 import UIKit
 
-func userNotificationCenterTap(userInfo: UNNotificationResponse) {
-    let userInf = userInfo.notification.request.content.userInfo
+class NotificationEvents {
+
+    @objc func userNotificationCenterTap(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
+        let userInfo = response.notification.request.content.userInfo
         let state = UIApplication.shared.applicationState
         if state == .inactive || state == .background {
-            if let aps = userInf["aps"] as? NSDictionary {
-                    if let alert = aps["alert"] as? NSDictionary {
-                        if let message = alert["message"] as? String {
+        if let aps = userInfo["aps"] as? NSDictionary {
+              if let alert = aps["alert"] as? NSDictionary {
+                     if let message = alert["message"] as? String {
```

Comment:
My reading of this code that this is uncontrolled text that may possibly include private/sensitive information.  We should look for something that gives a breakdown of the likely code path that generated or will follow from the push action; something that is code-level like a category or type identifier, and not actual message payload.

Reply:
ok, I will remove the message attribute which contains message payload information.
here we are capturing the screen name on notification click.

Reply:
The default UNNotificationResponse response is this.

[AnyHashable("aps"): {
    alert = "Splunk RUM.";
    badge = 1;
    sound = default;
}]